.github/workflows/linux.yml

```diff
@@ -87,6 +87,36 @@ jobs:
             -DAMReX_PARTICLES=ON
         make -j 2
 
+  test_sensei:
+    name: SENSEI Adaptor [test]
+    runs-on: ubuntu-18.04
+    env:
+      CXX: clang++
+      CC: clang
+      CXXFLAGS: "-Werror -Wshadow -Woverloaded-virtual -Wunreachable-code -fno-operator-names"
+      CMAKE_GENERATOR: Ninja
+      CMAKE_PREFIX_PATH: /root/install/sensei/develop/lib/cmake
+    container:
+      image: ryankrattiger/sensei:fedora33-vtk-mpi-20210616
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup
+      run: mkdir build
+    - name: Configure
+      run: |
+        cd build
+        cmake ..                  \
+            -DCMAKE_BUILD_TYPE=Debug \
+            -DAMReX_ENABLE_TESTS=ON  \
+            -DAMReX_FORTRAN=OFF      \
+            -DAMReX_PARTICLES=ON     \
+            -DAMReX_SENSEI=ON        \
+            -DCMAKE_CXX_STANDARD=14
```

Comment:
X-link https://github.com/ECP-WarpX/WarpX/pull/2210

The current need to pass this flag looks like a bug to me.
We use `target_compile_features(... PUBLIC cxx_std_14)` for the `amrex` target. Maybe a similar requirement is missing of an auxiliary SENSEI target in AMReX.

@c-wetterer-nelson @kwryankrattiger @burlen Can one of you please take this on?

Reply:
From what I could tell at the time I added this AMReX was using C++11 by default. In CMake, you should be able to just set the option globally `set(CMAKE_CXX_STANDARD 14)` if you want to use C++14 for the project.

I am not 100% on the project needs. Is there a reason you are setting it per target?

Reply:
Thanks @kwryankrattiger - yes, AMReX was C++11 at the time but now transitioned to C++14 as well.
For AMReX, we avoid setting `CMAKE_CXX_STANDARD` because the sets the C++ standard of the whole build - but downstream projects transition faster to newer C++ standards, thus `target_compile_features` is the better choice, which only declares a minimal requirement.

I tried removing `-DCMAKE_CXX_STANDARD=14`, but found it did not compile. Thus, there is somewhere a `target_compile_features` call missing in AMReX when SENSEI is built, I think.

Reply:
I think I found the reason for this: SENSEI overwrites our optimization flags and injects a C++ standard flag in the CXX flags:
```
/root/install/sensei/develop/lib/cmake/SENSEIConfig.cmake:  set(CMAKE_CXX_FLAGS "-fPIC -std=c++11 -Wall -Wextra -O3 -march=native -mtune=native"
```

We need to fix this upstream in SENSEI, we cannot inject these flags unconditionally into downstream builds:
https://github.com/SENSEI-insitu/SENSEI/blob/73873211eadc290991dcd62a5edddc45dc09d2bb/CMake/build.cmake#L18

Reply:
Upstream bug filed in https://github.com/SENSEI-insitu/SENSEI/issues/15